addons/mail/models/res_users.py

```diff
@@ -26,6 +29,14 @@ class Users(models.Model):
              "- Handle by Emails: notifications are sent to your email address\n"
              "- Handle in Odoo: notifications appear in your Odoo Inbox")
 
+    forced_im_status = fields.Selection([
+        ('away', 'Away'),
+        ('do_not_disturb', 'Do Not Disturb'),
+        ('offline', 'Offline')],
+        'Forced IM Status')
+    custom_status = fields.Char('Custom Status')
+    reset_custom_status_datetime = fields.Datetime('Reset Custom Status Datetime')
```

Comment:
I would keep `im` in the name of those fields, because `res.users` is broad so it's important to see these status refer to messaging/discuss thanks to `im`